master/internal/rm/kubernetesrm/pods.go

```diff
@@ -1221,6 +1239,25 @@ func (p *pods) cleanUpPodHandler(podHandler *pod) error {
 	return nil
 }
 
+func (p *pods) handleGetSlotsRequest(agentID string) *apiv1.GetSlotsResponse {
+	agentResp := p.handleGetAgentRequest(agentID)
+	slots := agentResp.Agent.Slots
+	response := &apiv1.GetSlotsResponse{}
+	for _, slot := range slots {
+		response.Slots = append(response.Slots, slot)
+	}
+	return response
+}
+
+func (p *pods) handleGetSlotRequest(agentID string, slotID string) *apiv1.GetSlotResponse {
+	agentResp := p.handleGetAgentRequest(agentID)
+	slots := agentResp.Agent.Slots
+	slot := slots[slotID]
+	response := &apiv1.GetSlotResponse{}
+	response.Slot = slot
+	return response
```

Comment:
```suggestion
	return  &apiv1.GetSlotResponse{Slot: slot}
```

Reply:
this applies in a lot of places, but i'll stop commenting it everywhere and leave you to fix the others.